### ISSUE TITLE
[User Story]: Open an assurance case in read-only mode

### ISSUE THREAD

**Issue by cptanalatriste** [alan-turing-institute/AssurancePlatform#565]
### Role

As an assurance case builder

### Desired Feature

I want to be able to view assurance cases shared with me, in read-only mode

### Benefit

So that I can work collaboratively on the development or review of an assurance case.

### Acceptance Criteria

**GIVEN** that another user has shared an assurance case with me, in read-only mode
**WHEN** I access this case, from the "Shared with me" section
**THEN** I cannot edit the description of any element
**AND** I cannot add or remove new elements
**AND** I cannot move, attach, re-attach elements
**AND** I cannot export the case to JSON.

### Technical Notes
NA

### Definition of Done

- [ ] The feature fulfills all acceptance criteria.
- [ ] The code is reviewed and meets team coding standards.
- [ ] All unit, integration, and end-to-end tests are passed.
- [ ] The feature meets accessibility standards set for the project.
- [ ] The code passes the CI/CD pipeline quality controls (style and linting).
- [ ] The feature is merged into the main branch.
- [ ] The feature is deployed and functional in the QA environment.
- [ ] The feature was demonstrated to the team and accepted by the product owner.

**Comment by cptanalatriste** [comment]
@chrisdburr , please have a look, and check if this is the feature you requested.

**Comment by chrisdburr** [comment]
Perhaps it would be helpful to agree on more fine-grained roles and permissions:

| **Role** | **Share Assurance Case** | **Manage Assurance Case (e.g. rename/delete)** | **Edit Assurance Case (e.g. add/edit elements or branches)** | **Add Comments** | **Download** | **View Assurance Case** |
| --- | --- | --- | --- | --- | --- | --- |
| **Owner** | x | x | x | x | x | x |
| **Editor** |  |  | x | x | x | x |
| **Reviewer** |  |  |  | x | x | x |
| **Viewer** |  |  |  |  |  | x |

This would allow us to be more specific with our user stories.

**Comment by chrisdburr** [comment]
So, then, for this particular user story, we could just focus on the assurance case `viewer`.

**Comment by cptanalatriste** [comment]
@chrisdburr , we have the Editor role defined in this user story (review pending) : https://github.com/alan-turing-institute/AssurancePlatform/issues/566

We have not covered the reviewer role yet, maybe you can create a user story?

**Comment by RichGriff** [comment]
I think I have covered most things here for the `Read Only` view of an Assurance Case `Shared with me`.

https://github.com/user-attachments/assets/319afc4d-c9d8-4429-ba09-f507ac6717cb

Just waiting on `permissions` property to be passed down when fetching cases - so we can handle the `delete` button when hovering over the card.

@chrisdburr @cptanalatriste can you please both review, thanks. 

Fyi not in `staging` as waiting on the property above.

**Comment by chrisdburr** [comment]
Looks good, @RichGriff. Please could I request a small banner that states the user is in read-only mode?

Not sure whether it should be in the sidebar or the main case window. Happy to leave this design choice to you based on what you think is best.

**Comment by RichGriff** [comment]
@chrisdburr I dont think a banner is a good idea as it may get in the way and distract the user from other elements. 

I have instead decided to so a lock icon next to fields in forms to indicate that this is read only. Making the form field disabled would have greyed out the text and i thought that wasnt ideal either. 

![Image](https://github.com/user-attachments/assets/ec879fd8-c1cd-4cc0-9d57-27a6e0087aa4)

With the limited options on screen and these 'locked' fields should be enough for a user to know they are in read only mode, without explicitly spelling it out for them.

**Comment by RichGriff** [comment]
@chrisdburr @cptanalatriste this is now in staging

**Comment by chrisdburr** [comment]
> @chrisdburr I dont think a banner is a good idea as it may get in the way and distract the user from other elements.
> 
> I have instead decided to so a lock icon next to fields in forms to indicate that this is read only. Making the form field disabled would have greyed out the text and i thought that wasnt ideal either.
> 
> ![Image](https://private-user-images.githubusercontent.com/28445922/359182545-ec879fd8-c1cd-4cc0-9d57-27a6e0087aa4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNTEzMDEsIm5iZiI6MTcyNDE1MTAwMSwicGF0aCI6Ii8yODQ0NTkyMi8zNTkxODI1NDUtZWM4NzlmZDgtYzFjZC00Y2MwLTlkNTctMjdhNmUwMDg3YWE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDEwNTAwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1M2QwYTExYjc0MDdjZTcxYjIyZWQ4MDA2NWIxM2VlODUxMDE2ZjFlM2U4ODUzMGRmYjVhOTY3NDA1M2NhZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xjovZSzpi6N_fljgiUX9hL4qVDTA8U7hz1Yde5QXyak)
> 
> With the limited options on screen and these 'locked' fields should be enough for a user to know they are in read only mode, without explicitly spelling it out for them.

This makes sense. Could we add a simple alt-text option to the lock though to explicitly state 'Read-only'? Just to really make clear that it's not a lock that can be unlocked.

**Comment by chrisdburr** [comment]
Closing issue. Feature verified in staging.
